pyuvsim/telescope.py

```diff
@@ -37,3 +40,243 @@ def __eq__(self, other):
                 and self.name == other.name
             )
         return False
+
+
+class BeamList(object):
+    """
+    A container for the set of beam models and related parameters.
+
+    Each rank in the simulation gets a copy of the set of beam objects
+    used for calculating Jones matrices. Rather than broadcasting the objects
+    themselves, the list is composed of strings which provide either a complete
+    description of an analytic beam or the path to a beamfits file. After the
+    broadcast, the beams are initialized from these strings.
+
+    This class provides methods to transform between the string and object
+    representations, and also carries parameters used globally by all UVBeams,
+    including their frequency / angular interpolation options.
```

Comment:
I think, based on code below, that these parameters are *required* to match between different UVBeams. If that's the case, that needs to be made more explicit. What happens in practice if you pass in UVBeam objects with different settings for these parameters and don't set a global override?

Reply:
I think if you make the list from objects that don't match you want to get a very clear error or warning message.

Reply:
That is essentially the behavior I was going for, but in hindsight I think there's a better option. The issue is that different parameters among UVBeams is lost when converted to strings, since those parameters are stored collectively in the `uvb_params` dictionary.

A safer option might be to allow for mismatches among UVBeams when in object mode, and raise an error in that case when converting to string mode. I'll have it raise a warning when adding a non-matching UVBeam to a list of objects. I'm not sure what to do when adding an UVBeam in string mode... that UVBeam's parameters will either have to overwrite the existing `uvb_params` or will be lost.